### PR TITLE
add prometheus metrics endpoint and provide billing metrics

### PR DESCRIPTION
**Pull request by jtlisi**

### CODE REVIEW

File: cmd/tsdb-gw/main.go [hunk 1]
```diff
@@ -189,3 +195,29 @@ func initRoutes(a *api.Api, enforceRoles bool) {
 	a.Router.Any("/prometheus/write", a.GenerateHandlers("write", enforceRoles, false, ingest.PrometheusMTWrite)...)
 	a.Router.Post("/metrics/delete", a.GenerateHandlers("write", enforceRoles, false, metrictank.MetrictankProxy("/metrics/delete"))...)
 }
+
+type metricsServer struct {
```
Comment (gouthamve) [inline cmd/tsdb-gw/main.go]: Hmm, this exists in `cortex-gw` also, can we put this in util?

File: publish/publish.go [hunk 2]
```diff
@@ -28,6 +38,10 @@ func Init(p Publisher) {
 }
 
 func Publish(metrics []*schema.MetricData) error {
+	if len(metrics) < 1 {
```
Comment (gouthamve) [inline publish/publish.go]: Hmm, `== 0` is a more common pattern I think.

File: publish/publish.go [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
 	"github.com/raintank/tsdb-gw/metrics_client"
 	log "github.com/sirupsen/logrus"
 	schema "gopkg.in/raintank/schema.v1"
 )
 
+var (
+	ingestedMetrics = promauto.NewCounter(prometheus.CounterOpts{
+		Namespace: "gateway",
+		Name:      "samples_ingested_total",
+		Help:      "Number of samples datapoints ingested",
```
Comment (Dieterbe) [inline publish/publish.go]: what's a samples datapoints ?

**Comment by Dieterbe** [comment]
don't forget to update ./scripts/config/tsdb-gw.ini

**Comment by jtlisi** [comment]
@gouthamve Each metrictank tenant has their own gateway so there is not need for a tenant id.

File: cmd/tsdb-gw/main.go (hunk 1 above)
Comment (jtlisi) [reply cmd/tsdb-gw/main.go]: Will do

File: publish/publish.go (hunk 2 above)
Comment (jtlisi) [reply publish/publish.go]: :+1:

File: publish/publish.go (hunk 3 above)
Comment (jtlisi) [reply publish/publish.go]: Each sample is a <value,timestamp>. We are just going to rate and sum across all the gateways for a org to get their ingestion rate.

File: scripts/config/tsdb-gw.ini [hunk 4]
```diff
@@ -55,3 +55,6 @@ tracing-addr = localhost:6831
 
 # limitations
 timerange-limit =
+
+# prometheus instrumentation
+metrics-addr = localhost:8001
```
Comment (Dieterbe) [inline scripts/config/tsdb-gw.ini]: should match the default value

Comment (jtlisi) [reply scripts/config/tsdb-gw.ini]: :+1:

**Pull request merged**